File: backend/app/schemas/config_center.py
from datetime import datetime
from pydantic import BaseModel, ConfigDict, Field


class DraftResponse(BaseModel):
    draft_id: str
    base_revision: int
    status: str
    modules: dict = Field(default_factory=dict)
    updated_at: datetime


class UpdateDraftModuleRequest(BaseModel):
    model_config = ConfigDict(extra="forbid")
    payload: dict = Field(default_factory=dict)
    operator: str | None = None


class ValidationIssue(BaseModel):
    field: str
    message: str


class ValidateDraftResponse(BaseModel):
    valid: bool
    errors: list[ValidationIssue] = Field(default_factory=list)
    warnings: list[ValidationIssue] = Field(default_factory=list)
    # FIX: [2026-08-22 P1] 原声明 list[str]，但 config_validator_service 对嵌套模块
    # 产生的 hints 为 {"field","message"} dict 列表（与 errors/warnings 同构），
    # pydantic 无法将 dict 校验为 str → ResponseValidationError → 接口 500。
    # 修正为 list[ValidationIssue]，与 service 实际返回结构一致。
    hints: list[ValidationIssue] = Field(default_factory=list)

File: frontend/src/api/configCenter.ts
import api from '@/utils/http'

export type DraftResponse = {
  draft_id: string
  base_revision: number
  status: string
  modules: Record<string, unknown>
  updated_at: string
}

export type ValidateResponse = {
  valid: boolean
  errors: Array<{ field: string; message: string }>
  warnings: Array<{ field: string; message: string }>
  hints: string[]
}

export const getCurrentDraft = async (): Promise<DraftResponse> => {
  const res = await api.get('/api/v1/config-center/drafts/current')
  return res.data
}

export const updateDraftModule = async (
  draftId: string,
  moduleName: string,
  moduleData: Record<string, unknown>
): Promise<DraftResponse> => {
  // FIXED: A-13 原来用 { payload } shorthand 导致请求体为 { payload: moduleData }
  // 后端 UpdateDraftModuleRequest schema 期望 { payload: dict, operator?: string }
  // 显式构造请求体，与后端 schema 字段名对齐，避免参数名混淆
  const res = await api.put(`/api/v1/config-center/drafts/${draftId}/modules/${moduleName}`, {
    payload: moduleData
  })
  return res.data
}

export const validateDraft = async (draftId: string): Promise<ValidateResponse> => {
  const res = await api.post(`/api/v1/config-center/drafts/${draftId}/validate`)
  return res.data
}
